posts/redux-sagas-all-you-need-to-know.md

```diff
@@ -0,0 +1,107 @@
+
+Sagas also have a learning curve, and if you’re only comfortable working with Redux or Context API, then you may not want to add another layer of complexity to your code.
+
+Finally, when diagnosing Sagas, the different architecture can lead to confusion about where the primary action is taking place and can make debugging more difficult if you’re unfamiliar with the architecture and ES6 generators.
+ 
+## Example Scenarios of Redux Sagas
+Let’s consider a scenario where you need to make an API request that depends on information from a previous API request in order to run. Trying to chain promises can lead to complex code that is challenging to test and maintain. Sagas can take this entire asynchronous operation and manage it in a simpler, more flexible way, by using a generator function to coordinate the three async operations.
+
+In a nutshell, an ES6 generator can be exited and re-entered at a later stage, with their context saved across instances.Using the next method will step through the function until it finds the next instance of the yield expression.
+
+Another scenario involves triggering actions only when certain conditions are met. Sagas has a helpful method called “take” that listens for specific actions to trigger by using only one loop.
+
+Read more on generator function [here](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Statements/function*).
+
+![Example Generator](https://raw.githubusercontent.com/Nero-Link/blog-usa/master/images/2023/04/generator%20example.PNG)
+
+Here are some example scenarios for when you might want to use Redux Sagas in your app:
+- When you need to make an API request and need to handle the response (for example, show a spinner while the request is being made)
+- When making API calls that require complex error handling and retries
+- When handling authentication flows that involve multiple API calls and redirects
+- When you need to manage a real-time connection between a client and a server (such as chat applications)
+- When managing data synchronisation between multiple sources, such as databases or web services
+- When creating animations or transitions that depend on a sequence of events
+- When handling long-running background processes, file uploads or other tasks
+
+Some other real-life applications of these examples:
+
+User authentication: Use Sagas to handle user authentication, which involves making API requests, storing tokens in local storage, and redirecting the user based on their authentication status.
+
+Real-time updates: Use Sagas to handle real-time updates in your app, such as notifications, chat messages, or live data updates from sensors or IoT devices.
+
+Multi-step form submission: Use Sagas to handle multi-step forms that involve multiple API requests, validation, error handling, and progress tracking.
```

Comment:
These examples are covered by some of the dot points above. I'd probably remove the dot points, and just slightly expand the real-life examples section